[docs] Add CLI uninstall flow to README.md

SamiSousa opened the pull request:
/cc @aravindhp

SamiSousa commented:
/cc @kevinrizza

kevinrizza commented on README.md:
````diff
@@ -124,3 +124,41 @@ operator-sdk test local ./test/e2e --up-local --kubeconfig=$KUBECONFIG --namespa
+    $ oc delete clusterserviceversion descheduler.v0.0.1 -n openshift-operators
+```
+
+Please note that we are working toward adding an uninstall flow as part of the OperatorHub UI.
````
I don't think adding this disclaimer is necessary.

SamiSousa replied:
I'll remove it

SamiSousa commented:
@kevinrizza Addressed your comment, thank you!

kevinrizza commented:
/lgtm

aravindhp commented on README.md:
```diff
@@ -124,3 +124,39 @@ operator-sdk test local ./test/e2e --up-local --kubeconfig=$KUBECONFIG --namespa
 
 You can also run the tests with `make e2e-test`.
+
+## Uninstalling an Operator from the CLI
```
This should go below the `Installing an operator using Marketplace` section

aravindhp commented on README.md:
```diff
@@ -124,3 +124,39 @@ operator-sdk test local ./test/e2e --up-local --kubeconfig=$KUBECONFIG --namespa
+
+After an operator has been installed from marketplace, to uninstall the operator from a particular OperatorGroup you need to delete the following resources. Below we are using the descheduler operator as an example:
+
+1. Delete the installation `CatalogSourceConfig` in the `openshift-marketplace` that contains the installed operator package. If created via the OperatorHub UI, it will be named `installed-<OPERATORSOURCE>-<TARGETNAMESPACE>` based on the `OperatorSource` that the operator comes from and the `targetNamespace` that the operator was installed to.
```
This should be the last step. Start with if there are multiple operators installed.

Please say `OpenShift OperatorHub UI`.

aravindhp commented on README.md:
````diff
@@ -124,3 +124,39 @@ operator-sdk test local ./test/e2e --up-local --kubeconfig=$KUBECONFIG --namespa
+
+1. Delete the installation `CatalogSourceConfig` in the `openshift-marketplace` that contains the installed operator package. If created via the OperatorHub UI, it will be named `installed-<OPERATORSOURCE>-<TARGETNAMESPACE>` based on the `OperatorSource` that the operator comes from and the `targetNamespace` that the operator was installed to.
+```bash
+    $ oc delete catalogsourceconfig installed-community-openshift-operators -n openshift-marketplace
````
Please use `kubectl`

aravindhp commented on README.md:
````diff
@@ -124,3 +124,39 @@ operator-sdk test local ./test/e2e --up-local --kubeconfig=$KUBECONFIG --namespa
+      csPublisher: Community
+      packages: descheduler,jaeger
+```
+Remove `descheduler` and the following comma:
````
You can leave out the verbiage about the comma.

aravindhp commented on README.md:
````diff
@@ -124,3 +124,39 @@ operator-sdk test local ./test/e2e --up-local --kubeconfig=$KUBECONFIG --namespa
+      packages: jaeger
+```
+Save the change and the marketplace-operator will reconcile the `CatalogSourceConfig`.
+2. Delete the `Subscription` in the `targetNamespace` that the operator was installed into. If created via the OperatorHub UI, it will be named after the operator's packageName.
````
Please account for how the subscription was created in `Installing an operator using Marketplace`

aravindhp commented on README.md:
````diff
@@ -124,3 +124,39 @@ operator-sdk test local ./test/e2e --up-local --kubeconfig=$KUBECONFIG --namespa
+3. Delete the `ClusterServiceVersion` in the `targetNamespace` that the operator was installed into. This will also delete the operator deployment, pod(s), rbac, and other resources that OLM created for the operator. This also deletes any corresponding CSVs that OLM "Copied" into namespaces watched by the operator.
+```bash
+    $ oc delete clusterserviceversion descheduler.v0.0.1 -n openshift-operators
+```
````
What about the `InstallPlan`? Does it get deleted when the `Subscription` gets deleted. If so please mention that.

SamiSousa replied:
I'm not entirely sure about the `InstallPlan`. I believe it should be deleted automatically if there's only one operator installed.
@ecordell how should we remove an operator from an `InstallPlan` with multiple operators/`Subscriptions`/CSVs?

tkashem commented:
/lgtm

SamiSousa commented:
@aravindhp Addressed your comments and have updated the PR.
Not sure how to address `InstallPlans`; @ecordell are there any docs about unsubscribing from operators in OLM?

SamiSousa commented:
/retest

tkashem commented:
/lgtm

SamiSousa commented:
/retest

awgreene commented:
/test e2e-aws-upgrade

aravindhp commented:
/retest
